scifin/timeseries/timeseries.py

```diff
@@ -200,9 +200,25 @@ def is_sampling_uniform(self):
             if intervals[i] - prev > 1.e-6:
                 return False
         return True
-    
 
-    
+
+PANDAS_METHODS = ["__add__", "__mul__", "__getitem__"]
+for name in PANDAS_METHODS:
+    def create_method(method_name):
+        def method(self, *args, **kwargs):
+            pd_series = self.data
+            func = getattr(pd_series, method_name)
+            new_args = []
+            for arg in args:
+                # if one of the args is a series turn into a pd.Series
+                if isinstance(arg, Series):
```

Comment:
Very clean implementation. If I understand correctly these methods apply to timeseries.Series(), hence to its derived classes. I see the great potential for timeseries.TimeSeries, but I don't know what __add__ and __mul__ would give for timeseries.CatTimeSeries. I think the 3 methods should be tested for class CatTimeSeries.